src/cfnlint/rules/resources/ResourceSchema.py

```diff
@@ -0,0 +1,29 @@
+"""
+Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+SPDX-License-Identifier: MIT-0
+"""
+import re
+from jsonschema import validate, ValidationError
+from cfnlint.helpers import REGEX_DYN_REF, PSEUDOPARAMS, FN_PREFIX, UNCONVERTED_SUFFIXES, REGISTRY_SCHEMAS
+from cfnlint.rules import CloudFormationLintRule
+from cfnlint.rules import RuleMatch
+
+class ResourceSchema(CloudFormationLintRule):
+    id = 'E3000'
+    shortdesc = 'Resource schema'
+    description = 'CloudFormation Registry resource schema validation'
+    source_url = 'https://github.com/aws-cloudformation/aws-cloudformation-resource-schema/'
+    tags = ['resources']
+
+    def match(self, cfn):
+        matches = []
+        for schema in REGISTRY_SCHEMAS:
+            resource_type = schema['typeName']
+            for resource_name, resource_values in cfn.get_resources([resource_type]).items():
+                properties = resource_values.get('Properties', {})
+                if not re.match(REGEX_DYN_REF, str(properties)) and not any(x in str(properties) for x in PSEUDOPARAMS + UNCONVERTED_SUFFIXES) and FN_PREFIX not in str(properties):
```

Comment:
line 24 is hard to read for me. what is `REGEX_DYN_REF`? 

`not any(x in str(properties) for x in PSEUDOPARAMS + UNCONVERTED_SUFFIXES) and FN_PREFIX not in str(properties)` 

can this be assigned to a variable `no_intrinsic_function_errors`?

Reply:
> what is `REGEX_DYN_REF`?

[dynamic references](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/dynamic-references.html):
https://github.com/aws-cloudformation/cfn-python-lint/blob/f90768da4513a5ee8d8782d86bc2256419f5f209/src/cfnlint/helpers.py#L77-L79

> can this be assigned to a variable `no_intrinsic_function_errors`?

added a comment above that line:
https://github.com/aws-cloudformation/cfn-python-lint/blob/82a5381b9e6ee375af164e8ae1dc1d74f183d63d/src/cfnlint/rules/resources/ResourceSchema.py#L24